runtime/jvmti/jvmtiThread.c

```diff
@@ -109,7 +109,7 @@ jvmtiGetThreadState(jvmtiEnv *env,
 				} else {
 					vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
 					/* The VM thread can't be recycled because getVMThread() prevented it from exiting. */
-					rv_thread_state = getThreadState(currentThread, targetThread->threadObject);
+					rv_thread_state = getThreadState(currentThread, targetThread->carrierThreadObject);
```

Comment:
`carrierThreadObject` only exists in Java19+. We will need to `ifdef` the changes as follows:
```suggestion
					rv_thread_state = getThreadState(
									currentThread,
#if JAVA_SPEC_VERSION >= 19
									targetThread->carrierThreadObject);
#else /* JAVA_SPEC_VERSION >= 19 */
									targetThread->threadObject);
#endif /* JAVA_SPEC_VERSION >= 19 */
```
Otherwise, JDK8, 11 and 17 builds will fail.

Reply:
added ifdef JAVA_SPEC_VERSION >= 19 as suggested.